app/main.py

```diff
@@ -1,13 +1,25 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
     pass
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed" if [format_linter_error(error)
+                               for error in errors] else "passed"
```

Comment:
why do you need to make a new list comprehension where you call `format_linter_error` to check if there are errors?
This function just formats errors in an appropriate way, it doesn't decide whether there are errors or not. The source of errors is "errors" list that you get as a parameter